src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -124,7 +124,7 @@ private BoundExpression RewriteDeconstruction(BoundTupleExpression left, Convers
             {
                 rightParts = ((BoundTupleExpression)right).Arguments;
             }
-            else if (right.Kind == BoundKind.Conversion)
+            else if (right.Kind == BoundKind.Conversion && ((BoundConversion)right).Operand.Kind == BoundKind.ConvertedTupleLiteral)
```

Comment:
I've made the change to accept both BoundTupleLiteral and BoundConvertedTupleLiteral. It does seem more robust.
The other issue (whether all tuple literals should be upgraded to BoundConvertedTupleLiteral during conversion) can be separated and discussed in the other PR.